src/main/java/org/gridsuite/modification/server/modifications/ShuntCompensatorModification.java

```diff
@@ -0,0 +1,88 @@
+
+/**
+ * @author Seddik Yengui <Seddik.yengui at rte-france.com>
+ */
+
+public class ShuntCompensatorModification extends AbstractModification {
+    private final ShuntCompensatorModificationInfos modificationInfos;
+
+    public ShuntCompensatorModification(ShuntCompensatorModificationInfos shuntCompensatorModificationInfos) {
+        this.modificationInfos = shuntCompensatorModificationInfos;
+    }
+
+    @Override
+    public void apply(Network network, Reporter subReporter) {
+        ShuntCompensator shuntCompensator = network.getShuntCompensator(modificationInfos.getEquipmentId());
+        if (shuntCompensator == null) {
+            throw new NetworkModificationException(SHUNT_COMPENSATOR_NOT_FOUND,
+                    String.format("Shunt compensator %s does not exist in network", modificationInfos.getEquipmentId()));
+        }
+        VoltageLevel voltageLevel = network.getVoltageLevel(modificationInfos.getVoltageLevelId());
+
+        subReporter.report(Report.builder()
+                .withKey("shuntCompensatorModification")
+                .withDefaultMessage("Shunt Compensator with id=${id} modified :")
+                .withValue("id", modificationInfos.getEquipmentId())
+                .withSeverity(TypedValue.INFO_SEVERITY)
+                .build());
+
+        ModificationUtils.getInstance().applyElementaryModifications(shuntCompensator::setName, () -> shuntCompensator.getOptionalName().orElse("No value"), modificationInfos.getEquipmentName(), subReporter, "Name");
+
+        if (shuntCompensator.getModelType() == ShuntCompensatorModelType.LINEAR) {
+            applyModificationOnLinearModel(subReporter, shuntCompensator, voltageLevel);
+        }
+
+    }
+
+    private void applyModificationOnLinearModel(Reporter subReporter, ShuntCompensator shuntCompensator, VoltageLevel voltageLevel) {
+        List<Report> reports = new ArrayList<>();
+        ShuntCompensatorLinearModel model = shuntCompensator.getModel(ShuntCompensatorLinearModel.class);
+        var shuntCompensatorType = model.getBPerSection() > 0 ? ShuntCompensatorType.CAPACITOR : ShuntCompensatorType.REACTOR;
+
+        if (modificationInfos.getShuntCompensatorType() != null) {
+            reports.add(ModificationUtils.getInstance().buildModificationReport(shuntCompensatorType, modificationInfos.getShuntCompensatorType().getValue(), "Type"));
+            shuntCompensatorType = modificationInfos.getShuntCompensatorType().getValue();
+            if (modificationInfos.getQAtNominalV() == null) {
+                // we retrieve the absolute value of susceptance per section, then we determine the sign using the type
+                double bPerSectionAbsoluteValue = Math.abs(model.getBPerSection());
+                double newBPerSection = shuntCompensatorType == ShuntCompensatorType.CAPACITOR ? bPerSectionAbsoluteValue : -bPerSectionAbsoluteValue;
+                model.setBPerSection(newBPerSection);
```

Comment:
Maybe we should add a report here as well ?

Reply:
This is the case when the user change only the shunt compensator type. There is report for that above at line 68: 
reports.add(ModificationUtils.getInstance().buildModificationReport(shuntCompensatorType, modificationInfos.getShuntCompensatorType().getValue(), "Type"));

Reply:
Then we don't want to log that bPerSectionAbsoluteValue change its value from positive to negative ? Thats a functionnal question, maybe we should ask a PO ?